qcow2reader_test.go

```diff
@@ -159,6 +162,361 @@ func BenchmarkExtentsUnallocated(b *testing.B) {
 	}
 }
 
+func TestExtentsSome(t *testing.T) {
+	clusterSize := 64 * KiB
+	extents := []image.Extent{
+		{
+			Start:     0 * clusterSize,
+			Length:    1 * clusterSize,
+			Allocated: true,
+		},
+		{
+			Start:  1 * clusterSize,
+			Length: 1 * clusterSize,
+			Zero:   true,
+		},
+		{
+			Start:     2 * clusterSize,
+			Length:    2 * clusterSize,
+			Allocated: true,
+		},
+		{
+			Start:  4 * clusterSize,
+			Length: 96 * clusterSize,
+			Zero:   true,
+		},
+		{
+			Start:     100 * clusterSize,
+			Length:    8 * clusterSize,
+			Allocated: true,
+		},
+		{
+			Start:  108 * clusterSize,
+			Length: 892 * clusterSize,
+			Zero:   true,
+		},
+		{
+			Start:     1000 * clusterSize,
+			Length:    16 * clusterSize,
+			Allocated: true,
+		},
+		{
+			Start:  1016 * clusterSize,
+			Length: 8984 * clusterSize,
+			Zero:   true,
+		},
+	}
+	qcow2 := filepath.Join(t.TempDir(), "image")
+	if err := createTestImageWithExtents(qcow2, qemuimg.FormatQcow2, extents); err != nil {
+		t.Fatal(err)
+	}
+	t.Run("qcow2", func(t *testing.T) {
+		actual, err := listExtents(qcow2)
+		if err != nil {
+			t.Fatal(err)
+		}
+		if !slices.Equal(extents, actual) {
+			t.Fatalf("expected %v, got %v", extents, actual)
+		}
+	})
+	t.Run("qcow2 zlib", func(t *testing.T) {
+		qcow2Zlib := qcow2 + ".zlib"
+		if err := qemuimg.Convert(qcow2, qcow2Zlib, qemuimg.FormatQcow2, qemuimg.CompressionZlib); err != nil {
+			t.Fatal(err)
+		}
+		actual, err := listExtents(qcow2Zlib)
+		if err != nil {
+			t.Fatal(err)
+		}
+		if !slices.Equal(compressed(extents), actual) {
```

Comment:
If slices is not available, we can add add an implementation for []image.Extent, or use relect.DeepEqual.